plugins/awards-backend/src/service/router.ts

```diff
@@ -29,6 +31,32 @@ export interface RouterOptions {
   tokenManager: TokenManager;
 }
 
+function getS3Client(config: Config): S3Client {
+  const region = config.getString('awards.s3.region');
+
+  // for local dev
+  const endpoint = config.getOptionalString('awards.s3.endpoint');
+
+  // credentials
+  const accessKey = config.getOptionalString('awards.s3.accessKey');
+  const secretKey = config.getOptionalString('awards.s3.secretKey');
+  const sessionToken = config.getOptionalString('awards.s3.sessionToken');
```

Comment:
If we did leverage a library like https://github.com/tweedegolf/storage-abstraction we could change the configuration to accept [a single DSN-like URL parameter](https://github.com/tweedegolf/storage-abstraction?tab=readme-ov-file#configuration-url) (like `s3://key:secret@the-buck/path/to/object?region=auto`) thus allowing users to easily choose any support storage backend with minimal effort needed on our end to instantiate it.

Reply:
I implemented storage-abstraction but didn't go with the single dsn url parameter. i've always found those dsn like urls a bit confusing and don't want to expose storage-abstraction's config url through our config.

but if you take a look think we have a good balance with the new tool and our own builders.

Reply:
I agree, this is a great balance